product_variant_cost_price/models/product_product.py

```diff
@@ -55,6 +55,12 @@ def _set_standard_price(self, product, value):
 
     @api.model
     def create(self, values):
+        if 'product_tmpl_id' in values and 'standard_price' not in values:
+            template = self.env['product.template'].browse(
+                values.get('product_tmpl_id'))
+            values.update({
```

Comment:
when you create a "new" variant using, create_variants_ids method from product.template the taken price is always 0.0 not the one in the template, that way it creates 2 history lines.